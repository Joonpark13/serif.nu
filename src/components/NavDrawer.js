import React from 'react';
import PropTypes from 'prop-types';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import { Link } from 'react-router-dom';

/* istanbul ignore next */
function ListItemLink(props) {
  return <ListItem button component={Link} {...props} />;
}

function NavDrawer({ isOpen, closeFunc }) {
  return (
    <Drawer onClose={closeFunc} open={isOpen}>
      <List>
        <ListItemLink to="/" onClick={closeFunc}>
          <ListItemText primary="Serif.nu" />
        </ListItemLink>
        <ListItemLink to="/about" onClick={closeFunc}>
          <ListItemText primary="About" />
        </ListItemLink>
        <ListItemLink to="/faq" onClick={closeFunc}>
          <ListItemText primary="FAQ" />
        </ListItemLink>
        <ListItemLink to="/report" onClick={closeFunc}>
          <ListItemText primary="Report a Bug" />
        </ListItemLink>
        <ListItemLink to="/contact" onClick={closeFunc}>
          <ListItemText primary="Contact" />
        </ListItemLink>
      </List>
    </Drawer>
  );
}

NavDrawer.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  closeFunc: PropTypes.func.isRequired,
};

export default NavDrawer;
