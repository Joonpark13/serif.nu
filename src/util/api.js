/* eslint-disable import/prefer-default-export */
export function searchURL(query) {
  return `${process.env.API_URL}/search?term_id=4720&search_query=${query}`;
}

export function sectionsURL(termId, schoolAbbv, subjectAbbv, courseAbbv) {
  return `${process.env.API_URL}/sections?term_id=${termId}&school_abbv=${schoolAbbv}&subject_abbv=${subjectAbbv}&course_abbv=${courseAbbv}`;
}
