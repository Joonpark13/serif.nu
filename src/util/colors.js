export const northwesternPurple = '#4E2A84';
export const northwesternBrightOrange = '#FFC520';
export const northwesternRichBlack10 = '#D8D6D6';
