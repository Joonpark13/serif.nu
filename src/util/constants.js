/* eslint-disable import/prefer-default-export */
export const CURRENT_TERM = '4730';
