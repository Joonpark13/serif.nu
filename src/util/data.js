import termsData from 'data/terms.json';

/* eslint-disable import/prefer-default-export */
/* istanbul ignore next */
export const currentTerm = termsData.reduce((highestTerm, term) => {
  if (term.id > highestTerm.id) {
    return term;
  }
  return highestTerm;
});
/* eslint-enable import/prefer-default-export */
