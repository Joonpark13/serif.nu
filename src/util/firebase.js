import firebase from 'firebase/app';
import 'firebase/firestore';
import 'firebase/auth';

firebase.initializeApp({
  apiKey: 'AIzaSyDluAG6bnqleqJ3yjPvGdlb3RyZP3U5JIQ',
  authDomain: 'sans-serif-northwestern.firebaseapp.com',
  databaseURL: 'https://sans-serif-northwestern.firebaseio.com',
  projectId: 'sans-serif-northwestern',
  storageBucket: 'sans-serif-northwestern.appspot.com',
  messagingSenderId: '1078100635889',
});

export const db = firebase.firestore();
export const auth = firebase.auth();
