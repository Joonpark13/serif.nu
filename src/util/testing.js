/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';
import * as useSelector from './use-selector';

jest.mock('./use-selector');

// Helper function that returns a function which creates enzyme wrappers with default props
export function wrapperCreator(Component, defaultProps = {}) {
  let classes;
  return props => shallow(<Component classes={classes} {...defaultProps} {...props} />);
}

/* istanbul ignore next */
export function mockUseSelector(...args) {
  useSelector.default.mockReset();
  args.forEach(arg => useSelector.default.mockReturnValueOnce(arg));

  return useSelector.default;
}

/* istanbul ignore next */
export function mockUseDispatch() {
  const dispatchSpy = jest.fn();
  reactRedux.useDispatch.mockReturnValue(dispatchSpy);
  return dispatchSpy;
}
