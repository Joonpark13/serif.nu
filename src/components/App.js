import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import {
  getSchoolsRequest,
  fetchSearchIndex as fetchSearchIndexAction,
} from 'actions';
import { auth } from 'util/firebase';
import toJS from 'util/to-js';
import { CURRENT_TERM } from 'util/constants';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { withRouter, Route } from 'react-router-dom';
import TopBar from './TopBar';
import NavDrawer from './NavDrawer';
import HomePage from './Pages/HomePage';
import AboutPage from './Pages/AboutPage';
import FAQPage from './Pages/FAQPage';
import ReportPage from './Pages/ReportPage';
import ContactPage from './Pages/ContactPage';

export const styles = {
  calendar: {
    overflow: 'auto',
  },
};

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      navigationOpen: false,
    };

    this.toggleNav = this.toggleNav.bind(this);
  }

  componentDidMount() {
    const { fetchSchools, fetchSearchIndex } = this.props;

    auth.signInAnonymously(); // TODO handle sign in error using .catch()

    auth.onAuthStateChanged((user) => {
      if (user) {
        fetchSchools(CURRENT_TERM);
        fetchSearchIndex(CURRENT_TERM);
      }
    });
  }

  toggleNav() {
    this.setState(oldState => ({
      navigationOpen: !oldState.navigationOpen,
    }));
  }

  render() {
    const { classes } = this.props;
    const { navigationOpen } = this.state;

    return (
      <Fragment>
        <TopBar menuAction={this.toggleNav} />
        <NavDrawer isOpen={navigationOpen} closeFunc={this.toggleNav} />
        <Route
          exact
          path="/"
          render={/* istanbul ignore next */ props => <HomePage {...props} classes={classes} />}
        />
        {/* istanbul ignore next */}
        <Route path="/about" component={AboutPage} />
        {/* istanbul ignore next */}
        <Route path="/faq" component={FAQPage} />
        {/* istanbul ignore next */}
        <Route path="/report" component={ReportPage} />
        {/* istanbul ignore next */}
        <Route path="/contact" component={ContactPage} />
      </Fragment>
    );
  }
}

App.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  fetchSchools: PropTypes.func.isRequired,
  fetchSearchIndex: PropTypes.func.isRequired,
};

const mapDispatchToProps = {
  fetchSchools: getSchoolsRequest,
  fetchSearchIndex: fetchSearchIndexAction,
};

export { App as UnstyledApp };
const AppWithStyles = toJS(withStyles(styles)(App));
export default withRouter(connect(null, mapDispatchToProps)(AppWithStyles));
