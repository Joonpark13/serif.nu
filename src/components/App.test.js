import { wrapperCreator } from 'util/testing';
import { UnconnectedApp } from './App';
import NavDrawer from './NavDrawer';

jest.mock('images/facebook-logo.png');
jest.mock('images/hero-image.jpg');
jest.mock('images/amy.jpg');
jest.mock('images/helen.jpg');
jest.mock('images/facebook-logo.png');
jest.mock('images/joon.jpeg');
jest.mock('images/julia.jpg');
jest.mock('images/kevin.jpg');
jest.mock('images/madison.jpg');
jest.mock('images/surprised_student.svg');

describe('App', () => {
  const defaultProps = {
    fetchSchools: () => {},
  };
  const getWrapper = wrapperCreator(UnconnectedApp, defaultProps);

  it('should render correctly', () => {
    const fetchSchoolsMock = jest.fn();
    const wrapper = getWrapper({ fetchSchools: fetchSchoolsMock });

    expect(wrapper.get(0)).toMatchSnapshot();
    expect(fetchSchoolsMock).toHaveBeenCalled();
  });

  it('should toggle navigation', () => {
    const wrapper = getWrapper();
    wrapper.find(NavDrawer).props().closeFunc();

    expect(wrapper.find(NavDrawer).prop('isOpen')).toBe(true);
  });
});
