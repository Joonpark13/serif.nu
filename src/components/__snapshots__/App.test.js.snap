// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<React.Fragment>
  <ForwardRef
    menuAction={[Function]}
  />
  <NavDrawer
    closeFunc={[Function]}
    isOpen={false}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    component={[Function]}
    path="/about"
  />
  <Route
    component={[Function]}
    path="/faq"
  />
  <Route
    component={[Function]}
    path="/report"
  />
  <Route
    component={[Function]}
    path="/contact"
  />
</React.Fragment>
`;

exports[`App should toggle navigation 1`] = `
<React.Fragment>
  <ForwardRef
    menuAction={[Function]}
  />
  <NavDrawer
    closeFunc={[Function]}
    isOpen={true}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    component={[Function]}
    path="/about"
  />
  <Route
    component={[Function]}
    path="/faq"
  />
  <Route
    component={[Function]}
    path="/report"
  />
  <Route
    component={[Function]}
    path="/contact"
  />
</React.Fragment>
`;
